### PR TITLE
feat(RecycleFBA): Add metrics

### PR DESCRIPTION
**Pull request by dadepo**
The goal of this PR is to add the ability to collect metrics from the RecycleFBA.

TODO
- [ ] Clarify on the metrics we want to collect.
- [ ] How to label the metircs? Given there would be many instance of recyle fba
- [ ] Ensure metrics collection does not affect the performance of allocating.
- [ ] Add metrics and confirm it works on ec2.

### CODE REVIEW

File: src/utils/allocators.zig [hunk 1]
```diff
@@ -204,6 +210,17 @@ pub fn RecycleFBA(config: struct {
     };
 }
 
+pub const RecycleFBAMetrics = struct {
```
Comment (dadepo) [inline src/utils/allocators.zig:213]: @0xNineteen Your ideas on metrics that should be collected here will be appreciated.

File: src/utils/allocators.zig [hunk 2]
```diff
@@ -16,13 +18,16 @@ pub fn RecycleFBA(config: struct {
...
+        // for tracking metrics.
+        maybe_metrics: ?RecycleFBAMetrics = null,
 
         const Record = struct { is_free: bool, buf: [*]u8, len: u64 };
         const AllocatorConfig = struct {
             // used for the records array
             records_allocator: std.mem.Allocator,
             // used for the underlying memory for the allocations
             bytes_allocator: std.mem.Allocator,
+            maybe_metrics: ?RecycleFBAMetrics = null,
```
Comment (0xNineteen) [inline src/utils/allocators.zig:30]: this should take an optional registry which is used to init the metrics (see how the account-cache in accountsdb works)

File: src/utils/allocators.zig (hunk 1 above)
Comment (0xNineteen) [reply src/utils/allocators.zig:213]: i think thats a great starting point -- we could also add a historgram of the sizes that are free

**Comment by 0xNineteen** [comment]
gonna close this for now since it seems dead - can reopen if we want to continue tackling it

**Pull request closed without merging**